test/api/importcsv/api_test.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/konveyor/tackle2-hub/binding"
+	"github.com/konveyor/tackle2-hub/test/assert"
+)
+
+func TestImportCSV(t *testing.T) {
+	for _, r := range TestCases {
+		t.Run(r.FileName, func(t *testing.T) {
+
+			// Upload CSV.
+			inputData := api.ImportSummary{}
+			assert.Must(t, Client.FilePost(api.UploadRoot, r.FileName, &inputData))
+
+			// Since uploading the CSV happens asynchronously we need to wait for the upload to check Applications and Dependencies.
+			time.Sleep(time.Second * 3)
+
+			// Check list of Applications.
+			gotApps, _ := Application.List()
+			expectedApps := r.ExpectedApplications
+			if len(gotApps) != len(expectedApps) {
+				t.Errorf("Mismatch in number of imported Applications: Expected %d, Actual %d", len(expectedApps), len(gotApps))
+			} else {
+				for i, importedApp := range gotApps {
+					if expectedApps[i].Name != importedApp.Name {
+						t.Errorf("Mismatch in name of imported Application: Expected %s, Actual %s", expectedApps[i].Name, importedApp.Name)
+					}
+					if expectedApps[i].Description != importedApp.Description {
+						t.Errorf("Mismatch in description of imported Application: Expected %s, Actual %s", expectedApps[i].Description, importedApp.Description)
+					}
+					if expectedApps[i].Repository.Kind != importedApp.Repository.Kind {
+						t.Errorf("Mismatch in repository's kind ofimported Application: Expected %s, Actual %s", expectedApps[i].Repository.Kind, importedApp.Repository.Kind)
+					}
+					if expectedApps[i].Repository.URL != importedApp.Repository.URL {
+						t.Errorf("Mismatch in repository's url of imported Application: Expected %s, Actual %s", expectedApps[i].Repository.URL, importedApp.Repository.URL)
+					}
+					if expectedApps[i].Binary != importedApp.Binary {
+						t.Errorf("Mismatch in binary of imported Application: Expected %s, Actual %s", expectedApps[i].Binary, importedApp.Binary)
+					}
+					for j, tag := range expectedApps[i].Tags {
+						if tag.Name != importedApp.Tags[j].Name {
+							t.Errorf("Mismatch in tag name of imported Application: Expected %s, Actual %s", tag.Name, importedApp.Tags[j].Name)
+						}
+					}
+					if expectedApps[i].BusinessService.Name != importedApp.BusinessService.Name {
+						t.Errorf("Mismatch in name of the BusinessService of imported Application: Expected %s, Actual %s", expectedApps[i].BusinessService.Name, importedApp.BusinessService.Name)
+					}
+				}
+			}
+
+			// Check list of Dependencies.
+			gotDeps, _ := Dependency.List()
+			expectedDeps := r.ExpectedDependencies
+			if len(gotDeps) != len(expectedDeps) {
+				t.Errorf("Mismatch in number of imported Dependencies: Expected %d, Actual %d", len(expectedDeps), len(gotDeps))
+			} else {
+				for i, importedDep := range gotDeps {
+					expectedDep := expectedDeps[i].To.Name
+					if importedDep.To.Name != expectedDep {
+						t.Errorf("Mismatch in imported Dependency: Expected %s, Actual %s", expectedDep, importedDep.To.Name)
```

Comment:
Is there a reason to not check also ```From``` side of the dependency? Consider adding it.

Reply:
Sure